### PR TITLE
[SYCL][CUDA] Move interop tests

### PR DESCRIPTION
**Pull request by bjoernknafla**
Move LIT test code for buffers that uses OpenCL C online compilation to
related interop test.

Apply clang-format.

Signed-off-by: Bjoern Knafla <bjoern@codeplay.com>

### CODE REVIEW

File: sycl/test/basic_tests/buffer/buffer_interop.cpp [hunk 1]
```diff
@@ -183,5 +182,86 @@ int main() {
     CHECK_OCL_CODE(clReleaseEvent(OpenCLEvent));
   }
 
+  {
+    queue myQueue;
```
Comment (alexbatashev) [inline sycl/test/basic_tests/buffer/buffer_interop.cpp:186]:
```suggestion
    queue MyQueue;
```

File: sycl/test/basic_tests/buffer/buffer_interop.cpp [hunk 2]
```diff
@@ -183,5 +182,86 @@ int main() {
...
+    if (!myQueue.is_host()) {
+      std::vector<int> data1(10, -1);
+      std::vector<int> data2(10, -2);
+      {
+        buffer<int, 1> a(data1.data(), range<1>(10));
+        buffer<int, 1> b(data2);
+
+        program prog(myQueue.get_context());
```
Comment (alexbatashev) [inline sycl/test/basic_tests/buffer/buffer_interop.cpp:194]:
```suggestion
        program Program(MyQueue.get_context());
```

File: sycl/test/basic_tests/buffer/buffer_interop.cpp [hunk 3]
```diff
@@ -183,5 +182,86 @@ int main() {
...
+        prog.build_with_source("kernel void override_source(global int* Acc) "
+                               "{Acc[get_global_id(0)] = 0; }\n");
+        cl::sycl::kernel krn = prog.get_kernel("override_source");
+        myQueue.submit([&](handler &cgh) {
```
Comment (alexbatashev) [inline sycl/test/basic_tests/buffer/buffer_interop.cpp:198]:
```suggestion
        MyQueue.submit([&](handler &CGH) {
```

File: sycl/test/basic_tests/buffer/buffer_interop.cpp (hunk 1 above)
Comment (bader) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp:186]: Is it still popular code style to use `my`/`My` prefix for variables names?

Comment (alexbatashev) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp:186]: It's hard to say. `q`, `Queue`, and `TestQueue` are also popular.

Comment (bader) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp:186]: > It's hard to say. `q`, `Queue`, and `TestQueue` are also popular.

IMO, any of these are better than `MyQueue`.

Comment (bjoernknafla) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp:186]: I only moved the code between tests without looking at it.  Will do the renaming to align with the surrounding code.

File: sycl/test/basic_tests/buffer/buffer_interop.cpp (hunk 2 above)
Comment (bjoernknafla) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp]: Done.

File: sycl/test/basic_tests/buffer/buffer_interop.cpp (hunk 3 above)
Comment (bjoernknafla) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp]: Done.

**Comment by bjoernknafla** [comment]
Also applied clang-format-9 to `buffer_interop.cpp`.

**Pull request merged**